.github/workflows/main.yml

```
@@ -0,0 +1,29 @@
+---
+name: Cargo Build & Test
+
+on:
+  push:
+  pull_request:
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build_and_test:
+    name: Rust project - latest
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        toolchain:
+          - stable
+          - beta
+          - nightly
+    steps:
+      - uses: actions/checkout@v4
+      - run: rustup update ${{ matrix.toolchain }} && rustup default ${{ matrix.toolchain }}
+      - run: rustup component add rustfmt clippy
+      - run: cargo build --verbose
+      - run: cargo fmt --all -- --check
+      - run: cargo clippy --all-targets -- -D warnings
+      - run: cargo test --verbose
+      
```

```suggestion
      - run: cargo test --verbose

```
Umm I think I made a mistake. My editor (or something) also inserted four spaces :(

---

Ok. Looks good now.